qiskit/dagcircuit/dagcircuit.py

```diff
@@ -216,7 +216,17 @@ def add_calibration(self, gate, qubits, schedule, params=None):
             Exception: if the gate is of type string and params is None.
         """
         if isinstance(gate, Gate):
-            self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
+            try:
+                gate_has_param_exp = isinstance(gate.params[0], ParameterExpression)
```

Comment:
There's no guarantee that because the 0th parameter is _not_ a `ParameterExpression` that another parameter won't be.

Reply:
Simplified the code so it's not dependent on `gate.params`.